File: angularapp/src/app/services/wifi-scheme-request.service.ts
import { Injectable } from '@angular/core';
import { WifiSchemeRequest } from '../models/wifi-scheme-request.model';
import { Observable } from 'rxjs/internal/Observable';
import { HttpClient, HttpHeaders } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class WifiSchemeRequestService {
  public apiUrl = 'https://8080-cfcebdbedeabbbdebbefcafbdaaafdbbeddebecdf.project.examly.io/api/wifiSchemeRequest';
  constructor(private http:HttpClient) { }
  private getAuthHeader(){
    return {
      headers:new HttpHeaders({Authorization: `Bearer ${localStorage.getItem('token')}`})
    };
  }

  getAllWifiSchemeRequests():Observable<WifiSchemeRequest[]>{
    return this.http.get<WifiSchemeRequest[]>(this.apiUrl,this.getAuthHeader());
  }
  getWifiSchemeRequestsById(id:number):Observable<WifiSchemeRequest>{
    return this.http.get<WifiSchemeRequest>(`${this.apiUrl}/${id}`,this.getAuthHeader());
  }
  addWifiSchemeRequest(data:WifiSchemeRequest):Observable<WifiSchemeRequest>{
    return this.http.post<WifiSchemeRequest>(this.apiUrl,data,this.getAuthHeader());
  } 
  updateWifiSchemeRequest(id:number, data:WifiSchemeRequest):Observable<WifiSchemeRequest>{
    return this.http.put<WifiSchemeRequest>(`${this.apiUrl}/${id}`,data, this.getAuthHeader());
  }
  deleteWifiSchemeRequestById(id:number):Observable<void>{
    return this.http.delete<void>(`${this.apiUrl}/${id}`,this.getAuthHeader());
  } 
}

File: springappwifi/src/main/java/com/examly/springappwifi/service/WiFiSchemeRequestService.java
package com.examly.springappwifi.service;

import java.util.List;
import java.util.Optional;

import com.examly.springappwifi.model.WiFiSchemeRequest;

public interface WiFiSchemeRequestService {
    WiFiSchemeRequest addWiFiSchemeRequest(WiFiSchemeRequest request);
    Optional <WiFiSchemeRequest> getWiFiSchemeRequestById(Long id);
    List<WiFiSchemeRequest> getAllWiFiSchemeRequests();
    WiFiSchemeRequest updateWiFiSchemeRequest(Long id, WiFiSchemeRequest request);
    boolean deleteWiFiSchemeRequest(Long id);
    List<WiFiSchemeRequest> getWiFiSchemeRequestsByUserId(Long userId);
}
